File: Python/Remove_Duplicates_Singly_LinkedList.py
class Node:
	def __init__(self, data):
		self.data = data
		self.next = None
		self.prev = None

class LinkedList:
	def __init__(self):
		self.head = None
			
	def push(self, data):
		newNode = Node(data)
		newNode.next = self.head
		self.head = newNode
		
	def printList(self, ptr):
		ptr = self.head
		if ptr is None:
			print('List has no element')
			return
		else:
			temp = ptr
			while(temp):
				print(temp.data, end=' ')
				temp = temp.next
				
	def removeDuplicates(self):
		ptr = self.head
		if ptr is None:
			return
		while ptr.next is not None:
			if ptr.data == ptr.next.data:
				temp = ptr.next.next
				ptr.next = temp
			else:
				ptr = ptr.next
			
llist = LinkedList()

llist.push(92)
llist.push(70)
llist.push(70)
llist.push(36)
llist.push(28)
llist.push(12)
llist.push(12)

print("Linked List before duplicate removal")
llist.printList(llist.head)

print("\n\nLinked List after duplicate removal")
result = llist.removeDuplicates()
llist.printList(llist.head)

File: C Language/Remove_Duplicates_Singly_LinkedList.c
//Remove Duplicates from Sorted Singly Linked List

#include<stdio.h>
#include<stdlib.h>

struct node
{
    int data;
    struct node * next;
    struct node * prev;
};

struct node *head = NULL;

void linkedListTraversal(struct node* ptr)
{
    ptr = head;
    while(ptr != NULL)
    {
        printf("%d ", ptr -> data);
        ptr = ptr -> next;
    }
}

void createList(int data)
{
    struct node *ptr = (struct node*)malloc(sizeof(struct node));
    ptr -> data = data;
    ptr -> next = head;	
    head = ptr;
}

void removeDuplicates(struct node* ptr)
{
    ptr = head;
    struct node* temp;
    if(ptr == NULL)
    {
        return;
    }
    while(ptr -> next != NULL)
    {
        if(ptr -> data == ptr -> next -> data)
        {
            temp = ptr -> next -> next;
            free(ptr -> next);
            ptr -> next = temp;
        }
        else
        {
            ptr = ptr -> next;
        }
    }
}


int main()
{
   
    createList(92);
    createList(70);
    createList(70);
    createList(36);
    createList(28);
    createList(28);
    createList(12);
    printf("Linked List before duplicate removal\n");
    linkedListTraversal(head);
    
    removeDuplicates(head);
    printf("\n\nLinked List after duplicate removal\n");
    linkedListTraversal(head);
    return 0;
}